agent/handlers/task_server_setup_integ_test.go

```diff
@@ -0,0 +1,135 @@
+//go:build integration
+// +build integration
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This function starts the server and listens on a specified port
+func startServer(t *testing.T) *http.Server {
+	router := mux.NewRouter()
+	registerFaultHandlers(router, nil, nil)
+	server := &http.Server{
+		Addr:    ":5932",
+		Handler: router,
+	}
+	go func() {
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			t.Logf("ListenAndServe(): %s\n", err)
+		}
+	}()
+	return server
+}
+
+// This function shuts down the server after the test
+func stopServer(t *testing.T, server *http.Server) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := server.Shutdown(ctx); err != nil {
+		t.Logf("Server Shutdown Failed:%+v", err)
+	} else {
+		t.Logf("Server Exited Properly")
+	}
+}
+
+// Table-driven tests for rate limiter
+func TestRateLimiterIntegration(t *testing.T) {
+
+	testCases := []struct {
+		name            string
+		method1         string
+		method2         string
+		url1            string
+		url2            string
+		expectedStatus2 int
+		assertNotEqual  bool
+	}{
+		{
+			name:            "Same network faults A1 + same methods B1",
+			method1:         "GET",
+			method2:         "GET",
+			url1:            "http://localhost:5932/api/container123/fault/v1/network-blackhole-port",
+			url2:            "http://localhost:5932/api/container123/fault/v1/network-blackhole-port",
+			expectedStatus2: http.StatusTooManyRequests,
```

Comment:
non-blocking: we can merge these two `expectedStatus2` and `assertNotEqual` to a single one since we only need to check if the second request is rate limited or not.